lib/CMakeLists.txt

```diff
@@ -4,10 +4,10 @@
 add_compile_options($<TARGET_PROPERTY:compiler,warning_shadow_variables>)
 
 add_subdirectory(crc)
-if(NOT CONFIG_EXTERNAL_LIBC)
```

Comment:
This change is even worse than the originally rejected one, as now the *internal* Zephyr libc code cmake code is included even if CONFIG_EXTERNAL_LIBC is set.
Which means one can both select CONFIG_EXTERNAL_LIBC and build a Zephyr provided library.

Reply:
This was your suggestion 😂

You were added to the PR.

![Screenshot_20231228-092727.png](https://github.com/zephyrproject-rtos/zephyr/assets/983494/9f3bf329-7f37-435f-aeef-8e43413af49b)

Reply:
This was not my suggestion. My suggestion is what is done in https://github.com/zephyrproject-rtos/zephyr/pull/67048